distribution/src/resources/drill-config.sh

```diff
@@ -193,21 +366,24 @@ if $is_cygwin; then
   DRILL_LOG_DIR=`cygpath -w "$DRILL_LOG_DIR"`
   CP=`cygpath -w -p "$CP"`
   if [ -z "$HADOOP_HOME" ]; then
-    HADOOP_HOME=${DRILL_HOME}/winutils
+    export HADOOP_HOME=${DRILL_HOME}/winutils
   fi
 fi
 
 # make sure allocator chunks are done as mmap'd memory (and reduce arena overhead)
-export MALLOC_ARENA_MAX=4
+# Newer versions of glibc use an arena memory allocator that causes virtual
+# memory usage to explode. Tune the variable down to prevent vmem explosion.
+export MALLOC_ARENA_MAX=${MALLOC_ARENA_MAX:-4}
 export MALLOC_MMAP_THRESHOLD_=131072
 export MALLOC_TRIM_THRESHOLD_=131072
 export MALLOC_TOP_PAD_=131072
 export MALLOC_MMAP_MAX_=65536
 
 # Variables exported form this script
-export HADOOP_HOME
 export is_cygwin
 export DRILL_HOME
 export DRILL_CONF_DIR
 export DRILL_LOG_DIR
 export CP
+export JAVA_HOME
+export JAVA
```

Comment:
The original ones are, well, original, so I let the as is. In the earlier version, drill-config.sh went to the trouble to compute JAVA_HOME and JAVA, but never exported them, making the work a bit of a waste. Now the values are exported.